google/cloud/security/common/gcp_api/_base_client.py

```diff
@@ -69,3 +71,41 @@ def _execute(self, request):
             upstream.
         """
         return request.execute()
+
+    def _build_paged_result(self, request, api_stub, rate_limiter):
+        """Execute results and page through the results.
+
+        Use of this method requires the API having a .list_next() method.
+
+        Args:
+            request: GCP API client request object.
+            api_stub: The API stub used to build the request.
+            rate_limiter: An instance of RateLimiter to use.
+
+        Returns:
+            A list of API response objects (dict).
+
+        Raises:
+            When the retry is exceeded, exception will be thrown.  This
+            exception is not wrapped by the retry library, and will be handled
+            upstream.
+
+            UnsupportedApiMethodError when the api_stub does not have the required
+            list_next() method.
+        """
+        if not hasattr(api_stub, 'list_next')
+            raise api_errors.UnsupportedApiMethodError
```

Comment:
A couple of small comments:
1: It might be nice to just raise ```api_errors.ApiExecutionError``` with the proper error message
```
     raise api_errors.ApiExecutionError(api_stub, 'No list_next() in API.)
```
2: by reusing ```api_errors.ApiExecutionError```, you don't have to catch it in admin_directory.py, and can just let this error bubble up to foo_pipeline.py, where it will be caught and handled.